### PR TITLE
feat(cart): add cart actions

### PR DESCRIPTION
**Pull request by griest024**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/graycoreio/daffodil/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

```
[ ] Bugfix
[x] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Other... Please describe:
```

## What is the current behavior?

The cart actions don't correspond to the current cart service interface.
Fixes: N/A

## What is the new behavior?
- Updates the actions to the new interface
- Adds generics on the actions where needed for the payload

## Does this PR introduce a breaking change?
```
[ ] Yes
[ ] No
```

## Other information

### CODE REVIEW

File: libs/cart/src/actions/cart.actions.ts [hunk 1]
```diff
@@ -3,27 +3,28 @@ import { Action } from '@ngrx/store';
...
+  CartLoadAction = '[DaffCart] Cart Load Action',
+  CartLoadSuccessAction = '[DaffCart] Cart Load Success Action',
+  CartLoadFailureAction = '[DaffCart] Cart Load Failure Action',
+  CartCreateAction = '[DaffCart] Cart Create Action',
+  CartCreateSuccessAction = '[DaffCart] Cart Create Success Action',
+  CartCreateFailureAction = '[DaffCart] Cart Create Failure Action',
+  AddToCartAction = '[DaffCart] Cart Add To Cart Action',
+  AddToCartSuccessAction = '[DaffCart] Cart Add to Cart Success Action',
+  AddToCartFailureAction = '[DaffCart] Cart Add to Cart Failure Action',
+  CartResetAction = '[DaffCart] Cart Reset Action',
```
Comment (lderrickable) [inline libs/cart/src/actions/cart.actions.ts]: Will this action trigger a clearCart API call? Either way, I think you're going to need a clear cart action

File: libs/cart/src/actions/cart.actions.ts [hunk 2]
```diff
@@ -3,27 +3,28 @@ import { Action } from '@ngrx/store';
...
+  CartResetSuccessAction = '[DaffCart] Cart Reset Success Action',
+  CartResetFailureAction = '[DaffCart] Cart Reset Failure Action'
 }
 
 export class DaffCartLoad implements Action {
```
Comment (lderrickable) [inline libs/cart/src/actions/cart.actions.ts]: So the idea here is to get the id from state/localstorage? Then make a createCart call if there is no id?

File: libs/cart/src/actions/cart.actions.ts [hunk 3]
```diff
@@ -32,46 +33,66 @@ export class DaffCartLoadFailure implements Action {
...
-    constructor(public payload: {productId: string, qty: number}) {}
+  constructor(public payload: {productId: string, qty: number}) {}
 }
 
 export class DaffAddToCartSuccess implements Action {
-    readonly type = DaffCartActionTypes.AddToCartSuccessAction;
+  readonly type = DaffCartActionTypes.AddToCartSuccessAction;
 
-    constructor(public payload: DaffCart) {}
+  constructor(public payload: DaffCart) {}
```
Comment (lderrickable) [inline libs/cart/src/actions/cart.actions.ts]: Should be generic

File: libs/cart/src/actions/cart.actions.ts (hunk 1 above)
Comment (griest024) [reply libs/cart/src/actions/cart.actions.ts]: Yeah I repurposed the reset action for the clear call but didn't rename the actions.

File: libs/cart/src/actions/cart.actions.ts (hunk 2 above)
Comment (griest024) [reply libs/cart/src/actions/cart.actions.ts]: Yeah that was my idea.

File: libs/cart/src/actions/cart.actions.ts [hunk 4]
```diff
@@ -32,46 +33,66 @@ export class DaffCartLoadFailure implements Action {
...
+
+  constructor(public payload: string) {}
+}
+
 export class DaffAddToCart implements Action {
-    readonly type = DaffCartActionTypes.AddToCartAction;
+  readonly type = DaffCartActionTypes.AddToCartAction;
 
-    constructor(public payload: {productId: string, qty: number}) {}
+  constructor(public payload: {productId: string, qty: number}) {}
```
Comment (lderrickable) [inline libs/cart/src/actions/cart.actions.ts]: seems inconsistent that the `DaffCartCreateSuccess` action takes a payload of `{id: T['id']}` while this takes a payload of `{productId: string, qty: number}`. Seems like it should be something like `{productId: T['id'], qty: T['quantity']}`

Comment (griest024) [reply libs/cart/src/actions/cart.actions.ts]: You're right, it is inconsistent. I didn't update anything related to the deprecated `addToCart` call.

Comment (lderrickable) [reply libs/cart/src/actions/cart.actions.ts]: oh, right. I forgot it was deprecated. Nevermind then.

File: libs/cart/src/actions/cart.actions.ts (hunk 2 above)
Comment (lderrickable) [reply libs/cart/src/actions/cart.actions.ts]: looks good then

File: libs/cart/src/actions/cart.actions.ts (hunk 1 above)
Comment (lderrickable) [reply libs/cart/src/actions/cart.actions.ts]: Do you think it would make your life much more difficult to rename them to "clear"?

Comment (griest024) [reply libs/cart/src/actions/cart.actions.ts]: Nope, will do.

**Pull request closed without merging**